powertools-metrics/src/main/java/software/amazon/lambda/powertools/metrics/internal/LambdaMetricsAspect.java

```diff
@@ -0,0 +1,84 @@
+import static software.amazon.lambda.powertools.core.internal.LambdaHandlerProcessor.isColdStart;
+import static software.amazon.lambda.powertools.core.internal.LambdaHandlerProcessor.isHandlerMethod;
+import static software.amazon.lambda.powertools.core.internal.LambdaHandlerProcessor.placedOnRequestHandler;
+import static software.amazon.lambda.powertools.core.internal.LambdaHandlerProcessor.placedOnStreamHandler;
+import static software.amazon.lambda.powertools.core.internal.LambdaHandlerProcessor.serviceName;
+import static software.amazon.lambda.powertools.metrics.PowertoolsMetricsLogger.metricsLogger;
+import static software.amazon.lambda.powertools.metrics.PowertoolsMetricsLogger.withSingleMetric;
+
+@Aspect
+public class LambdaMetricsAspect {
+    private static final String NAMESPACE = System.getenv("POWERTOOLS_METRICS_NAMESPACE");
+
+    @SuppressWarnings({"EmptyMethod"})
+    @Pointcut("@annotation(powertoolsMetrics)")
+    public void callAt(PowertoolsMetrics powertoolsMetrics) {
+    }
+
+    @Around(value = "callAt(powertoolsMetrics) && execution(@PowertoolsMetrics * *.*(..))", argNames = "pjp,powertoolsMetrics")
+    public Object around(ProceedingJoinPoint pjp,
+                         PowertoolsMetrics powertoolsMetrics) throws Throwable {
+        Object[] proceedArgs = pjp.getArgs();
+
+        if (isHandlerMethod(pjp)
+                && (placedOnRequestHandler(pjp)
+                || placedOnStreamHandler(pjp))) {
+
+            MetricsLogger logger = metricsLogger();
+
+            logger.setNamespace(namespace(powertoolsMetrics))
+                    .putDimensions(DimensionSet.of("service", service(powertoolsMetrics)));
+
+            coldStartSingleMetricIfApplicable(pjp, powertoolsMetrics);
+
+            Object proceed = pjp.proceed(proceedArgs);
+
+            coldStartDone();
+            logger.flush();
+            return proceed;
+        }
+
+        return pjp.proceed(proceedArgs);
+    }
+
+    private void coldStartSingleMetricIfApplicable(final ProceedingJoinPoint pjp,
+                                                   final PowertoolsMetrics powertoolsMetrics) {
+        if (powertoolsMetrics.captureColdStart()
+                && isColdStart()) {
+
+            Optional<Context> contextOptional = extractContext(pjp);
+
+            if (contextOptional.isPresent()) {
+                Context context = contextOptional.orElseThrow(() -> new IllegalStateException("Context not found"));
+
+                withSingleMetric("ColdStart", 1, Unit.COUNT, namespace(powertoolsMetrics), (logger) ->
+                        logger.setDimensions(DimensionSet.of("service", service(powertoolsMetrics)),
+                                DimensionSet.of("function_name", context.getFunctionName())));
+            }
+        }
+    }
+
+    private String namespace(PowertoolsMetrics powertoolsMetrics) {
+        return !"".equals(powertoolsMetrics.namespace()) ? powertoolsMetrics.namespace() : NAMESPACE;
```

Comment:
What is the default namespace if neither the annotation parameter or environment variable are set?

Reply:
This should probably match the "service_undefined" functionality.

Reply:
I dont see python version sets any defaults either https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/aws_lambda_powertools/metrics/base.py#L90